lib/validation.ts

```diff
@@ -187,7 +188,7 @@ export function fileExists(_path: string): boolean {
   return true;
 }
 
-export function checkAndConvertToJson(_path: string): object | null {
+export function checkAndConvertToJson<T>(_path: string): T | null {
```

Comment:
This generic seems to be functioning as an implicit type assertion since it is only used to specify the return type. I.E. is basically the same as `checkAndConvertToJson(somePath) as T`, which bypasses the type system entirely.

I think ideally we should make this method return `unknown` and then the consuming pieces of code would need to type narrow on their end using type predicates.

Reply:
I see that you added the type predicates which is great 👍 

Could we make this method return `unknown | null` and remove the generic?